modules/config.ts

```diff
@@ -136,24 +135,14 @@ class Config {
         //We check the most likely install paths.
         if (os.platform() == "win32") {
             const steamPaths = ["C:/Program Files (x86)/Steam", "C:/Program Files/Steam"];
-            //@ts-ignore
             basedir = await getExistingPath(steamPaths);
         }
         else if (os.platform() == "linux" || "freebsd" || "openbsd") {
             //Linux solution is untested
             const homedir = process.env.HOME;
-            const steamPaths = [
-                {
-                    "paths": [".steam/steam"],
-                    "prefix": homedir
-                },
-                {
-                    "paths": [".local/share/steam"],
-                    "prefix": ""
-                }
-            ];
+            const steamPaths = [".steam/steam", ".local/share/steam"];
             for (const pathGroup of steamPaths) {
-                basedir = await getExistingPath(pathGroup["paths"], pathGroup["prefix"]);
+                basedir = homedir + await getExistingPath(pathGroup["paths"]);
```

Comment:
I think using await in this way will cause an issue, so we should instead do:
```
let existingPath = await getExistingPath(pathGroup["paths"]);
basedir = path.join(honedir, existingPath);
```
Always best to path join for directory contatinations.

Reply:
pseudo code, plz double check!